### PR TITLE
Memory-pair-game

### PR DESCRIPTION
**Pull request by kasionio**
### memory-pair-game

[Demo](https://kasionio.github.io/memory-pair-game)
[Code base](https://github.com/kasionio/frontend-2021-homeworks/blob/memory-pair-game/submissions/kasionio/memory-pair-game/index.js)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/kasionio/memory-pair-game/index.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        if(!isCardFlipped(clickedCard)) {
+            cardPairsArray.push(target.id);
+        }
+        flip(clickedCard); 
+
+        if (cardPairsArray[0] !== cardPairsArray[1] && cardPairsArray.length === 2) {
+            pauseEventListener(cardsWrapper, checkCard, DELAY_TIME);
+            setTimeout(() => {
+                document.querySelectorAll('.flip-card_flipped').forEach(card => card.closest('.flip-card').classList.remove('flip-card_flipped'));
+            }, DELAY_TIME);
```
Comment (babayK0) [inline submissions/kasionio/memory-pair-game/index.js:75]: You made function `isCardFlipped` without which it was possible to do without, but here i suggest you to make separate function for this part.

File: submissions/kasionio/memory-pair-game/index.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            numberOfAttempts++;
+            cleanArr(cardPairsArray);    
+        }
+
+        else if (cardPairsArray[0] === cardPairsArray[1] && cardPairsArray.length === 2) { 
+            pauseEventListener(cardsWrapper, checkCard, DELAY_TIME);
+            document.querySelectorAll(`[id='${cardPairsArray[0]}']`).forEach(card => card.closest('.flip-card').classList.add('hide'));
```
Comment (babayK0) [inline submissions/kasionio/memory-pair-game/index.js:82]: Not obvious and not readable part of code. Move it to separate function

File: submissions/kasionio/memory-pair-game/index.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    }
+});
+
+function reloadGame() {
+    document.location.reload();
+}
+
+function flip(card) {
+    card.classList.add('flip-card_flipped');
+    card.style.transitionDuration = '0.5s';
```
Comment (babayK0) [inline submissions/kasionio/memory-pair-game/index.js:105]: use css for styles

**Review by babayK0 — CHANGES_REQUESTED** [review]
Nice job!
Watch my comments below please :smile_cat:

File: submissions/kasionio/memory-pair-game/index.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    },
+    {
+    id: 6,
+    src: 'img/6.jpg',
+    alt: 'cat'
+    }
+];
+
+const allCards = [...cards, ...cards];
+allCards.sort(() => { return 0.5 - Math.random() });
```
[the remark this answers is not in the document]
Comment (kasionio) [reply submissions/kasionio/memory-pair-game/index.js:35]: Thanks. Initialize new variable

File: submissions/kasionio/memory-pair-game/index.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+cardsWrapper.addEventListener('click', function checkCard({target}) {
+    const clickedCard = target.closest('.flip-card');
+    if (clickedCard !== null) {
+        if(!isCardFlipped(clickedCard)) {
+            cardPairsArray.push(target.id);
+        }
+        flip(clickedCard); 
+
+        if (cardPairsArray[0] !== cardPairsArray[1] && cardPairsArray.length === 2) {
```
[the remark this answers is not in the document]
Comment (kasionio) [reply submissions/kasionio/memory-pair-game/index.js:71]: Genius 💯 This makes code much better, thanks a lot!

File: submissions/kasionio/memory-pair-game/index.js (hunk 2 above)
Comment (kasionio) [reply submissions/kasionio/memory-pair-game/index.js:82]: Make code more ES6 looking

**Review by babayK0 — APPROVED** [review]
Good job !

**Pull request merged**